app.hypermedia.testing.dsl/src/main/java/app/hypermedia/testing/dsl/generator/CoreGenerator.xtend

```diff
@@ -34,116 +40,112 @@ class CoreGenerator extends AbstractGenerator {
         if ( ! blocks.empty) {
             val String dslFileName = resource.getURI().lastSegment.toString();
 
-            fsa.generateFile(dslFileName + '.json', generateSteps(blocks));
+            fsa.generateFile(dslFileName + '.json', generateSteps(blocks).toString(INDENTATION));
         }
     }
 
-    def generateSteps(Iterable<EObject> blocks) '''
-        {
-            "steps": [
-                «FOR block:blocks SEPARATOR ","»
-                    «block.step»
-                «ENDFOR»
-            ]
+    def generateSteps(Iterable<EObject> blocks) {
+        val scenario = new JSONObject();
+        val steps = new JSONArray()
+        
+        for (block : blocks) {
+            val stepJson = block.step
+        	steps.put(new JSONObject(stepJson.toString()))
         }
-    '''
+        
+        scenario.put('steps', steps)
+        return scenario
+    }
+    
+    def buildBlock(String type, Iterable<? extends EObject> children, Map<String, Object> map) {   
+        map.put('children', children.map([c | c.step as JSONObject]).toList())
```

Comment:
is casting `c.step as JSONObject` necessary?

Reply:
In fact it isn't. Xtend complained when `map` was called in a different place, but not here actually :+1:.